internal/extsvc/azuredevops/client.go

```
@@ -126,14 +118,31 @@ func (c *Client) do(ctx context.Context, req *http.Request, result any) (*http.R
 	return resp, json.Unmarshal(bs, result)
 }
 
+// WithAuthenticator returns a new Client that uses the same configuration,
+// HTTPClient, and RateLimiter as the current Client, except authenticated with
+// the given authenticator instance.
+//
+// Note that using an unsupported Authenticator implementation may result in
+// unexpected behaviour, or (more likely) errors. At present, only BasicAuth is
+// supported.
+func (c Client) WithAuthenticator(a auth.Authenticator) Client {
```

Thanks for adding this! 
I wonder if we should just throw an error if the type is not Basic, if all the others don't work today :D

---

I was thinking about that actually, but all of the other implementations did this at the Source level as opposed to the client. I think it makes sense to have it in both probably.

---

I wonder if it actually makes _more_ sense to have it in the client, since the source will just forward the authenticator to the client anyways and will keep the existing behavior, while having the safeguard in place for both.

---

Makes sense to me!

---

changes pushed!